e2e/test_operator/operator_test.go

```diff
@@ -1428,6 +1429,50 @@ var _ = Describe("Operator", Label("e2e", "pr"), func() {
 		})
 	})
 
+	// this test took 14min to run for me, so I think it is best to just have the one test
+	When("replacing Pods due to securityContext changes", func() {
+		var initialPods *corev1.PodList
+		var originalPodSpec, modifiedPodSpec *corev1.PodSpec
+
+		BeforeEach(func() {
+			if fdbCluster.GetClusterSpec().ReplaceInstancesWhenResourcesChange != nil {
+				Expect(*fdbCluster.GetClusterSpec().ReplaceInstancesWhenResourcesChange).To(BeFalse())
+			}
+			originalPodSpec = fdbCluster.GetPodTemplateSpec(fdbv1beta2.ProcessClassStorage)
+			Expect(originalPodSpec).NotTo(BeNil())
+			modifiedPodSpec = originalPodSpec.DeepCopy()
+			Expect(modifiedPodSpec.SecurityContext).NotTo(BeNil())
+			modifiedPodSpec.SecurityContext.FSGroupChangePolicy = &[]corev1.PodFSGroupChangePolicy{corev1.FSGroupChangeOnRootMismatch}[0]
+			initialPods = fdbCluster.GetStoragePods()
+			Expect(fdbCluster.SetPodTemplateSpec(fdbv1beta2.ProcessClassStorage, modifiedPodSpec, false)).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(fdbCluster.SetPodTemplateSpec(fdbv1beta2.ProcessClassStorage, originalPodSpec, true)).NotTo(HaveOccurred())
+		})
+
+		It("should replace all the storage pods (which had the securityContext change)", func() {
+			lastForcedReconciliationTime := time.Now()
+			forceReconcileDuration := 4 * time.Minute
+			Eventually(func(g Gomega) bool {
+				// Force a reconcile if needed to make sure we speed up the reconciliation if needed.
+				if time.Since(lastForcedReconciliationTime) >= forceReconcileDuration {
+					fdbCluster.ForceReconcile()
+					lastForcedReconciliationTime = time.Now()
+				}
+
+				// check that all original pods are gone
+				pods := fdbCluster.GetStoragePods()
+				g.Expect(pods.Items).NotTo(ContainElements(initialPods.Items))
+				for _, pod := range pods.Items {
```

Comment:
would it make more sense to instead check if all original storage pods/pgs are marked for removal? or would that be incomplete

Reply:
Have you tested what the difference in for the test execution would be? Will be the test much faster or still take that long?

Reply:
no, I wasn't sure if it was a valid way to test honestly, since the point is that the pod be replaced

Reply:
I'll give it a go

Reply:
772.232 seconds =~13min so not much improvement (this is also with storage pods) I'll do some tests with general / log